File: Java Programs/Chapter 1/BubbleSort/src/bubblesort/BubbleSort.java

package bubblesort;

//Program to sort the elements using bubble sort.

import java.io.*;
import java.util.*;

public class BubbleSort
{
	public static void main(String args[])
	{
		int n,temp,i,pass;
		int A[]=new int[20];
		Scanner oin=new Scanner(System.in);
		
		//Getting the elements in the array.
		System.out.print("Enter the number of elements in the array: ");
		n=Integer.parseInt(oin.nextLine());
			
		System.out.println("Enter the elements of the array:");
		for(i=1;i<=n;i++)
		{
			A[i]=Integer.parseInt(oin.nextLine());
		}
		
		//sort
		for(pass=1;pass<=n;pass++)
		{
			for(i=1;i<=(n-pass);i++)
			{
				if(A[i]>A[i+1])
				{
					temp=A[i];
					A[i]=A[i+1];
					A[i+1]=temp;
				}
			}
		}
		System.out.print("Sorted Array: ");
		for(i=1;i<=n;i++)
		{
			System.out.print(""+A[i]+" ");
		}
	}
}

File: Python Programs/unit1/BubbleSort.py
from __future__ import print_function


def Bubble_Sort():
    #    global array,n
    for i in range(0, n):
        for j in range(0, n):
            if(array[i] < array[j]):
                temp = array[i]
                array[i] = array[j]
                array[j] = temp


array = []
print("Enter the number of elements in an array")
n = int(raw_input())
print("Enter the elements")
for i in range(0, n):
    array.append(int(raw_input()))
print("Elements in the array")
for i in array:
    print("%d" % i)
Bubble_Sort()
print("Elements in sorted order")
for i in array:
    print("%d" % i)
